homeassistant/components/intellifire/sensor.py

```diff
@@ -0,0 +1,140 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Any
+
+from intellifire4py import IntellifirePollData
+import pytz
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IntellifireDataUpdateCoordinator
+from .const import DOMAIN
+
+
+class IntellifireSensor(CoordinatorEntity, SensorEntity):
+    """Define a generic class for Sensors."""
+
+    # Define types
+    coordinator: IntellifireDataUpdateCoordinator
+    entity_description: IntellifireSensorEntityDescription
+    _attr_attribution = "Data provided by unpublished Intellifire API"
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        description: IntellifireSensorEntityDescription,
+    ) -> None:
+        """Init the sensor."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+
+        # Set the Display name the User will see
+        self._attr_name = f"Fireplace {description.name}"
+        self._attr_unique_id = f"{description.key}_{coordinator.api.data.serial}"
+        # Configure the Device Info
+        self._attr_device_info = self.coordinator.device_info
+
+    @property
+    def native_value(self) -> int | str:
+        """Return the state."""
+        return self.entity_description.value_fn(self.coordinator.api.data)
+
+
+def _time_remaining_to_timestamp(data: IntellifirePollData) -> datetime | None:
+    """Define a sensor that takes into account timezone."""
+
+    seconds_offset = data.timeremaining_s
+    # If disabled return None - else return a timestamp with correct TZ info
+    if seconds_offset == 0:
+        return None
+
+    return datetime.now().replace(
+        tzinfo=pytz.timezone(self.hass.config.time_zone)
+    ) + timedelta(seconds=seconds_offset)
```

Comment:
Is it compulsory to use `self.hass.config.time_zone`?
You may need to adjust the lambda to be able to pass `hass`, or to pass the timezone as an argument.

Reply:
When I tried to return a `datetime` w/out the timezone info I was getting errors ... so I thought I had to figure a way to pass it in.

Reply:
I think you should also clear out at least the microseconds and miliseconds from the returned object.